src/wallet/rpcwallet.cpp

```diff
@@ -1334,12 +1334,26 @@ static void MaybePushAddress(UniValue & entry, const CTxDestination &dest)
         entry.push_back(Pair("address", addr.ToString()));
 }
 
-void ListTransactions(const CWalletTx& wtx, const string& strAccount, int nMinDepth, bool fLong, UniValue& ret, const isminefilter& filter)
+/**
+ * List transactions based on the given criteria. If the criteria applied in
+ * reference to being sent or received (if the ret UniValue array was appended
+ * to), the function returns true, otherwise false is returned.
+ *
+ * @param  wtx        The wallet transaction.
+ * @param  strAccount The account, if any, or "*" for all.
+ * @param  nMinDepth  The minimum confirmation depth.
+ * @param  fLong      Whether to include the JSON version of the transaction.
+ * @param  ret        The UniValue into which the result is stored.
+ * @param  filter     The "is mine" filter bool.
+ * @return            true if appends were made to ret, false otherwise.
+ */
+bool ListTransactions(const CWalletTx& wtx, const string& strAccount, int nMinDepth, bool fLong, UniValue& ret, const isminefilter& filter)
```

Comment:
For the case where tx0 is in reorged and main chain both (see `test_double_send`), both versions of the transaction will appear in the results (once in 'transactions' and once in 'replaced'). To prevent that, I check if `ListTransactions` appended to the transactions array -- if it did, I need to not include in replaced.